core/templates/pages/exploration-player-page/layout-directives/progress-nav.component.spec.ts

```diff
@@ -0,0 +1,143 @@
+      ],
+      providers: [
+        BrowserCheckerService,
+        ExplorationEngineService,
+        ExplorationPlayerStateService,
+        FocusManagerService,
+        PlayerPositionService,
+        PlayerTranscriptService,
+        UrlService,
+        WindowDimensionsService
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ProgressNavComponent);
+    componentInstance = fixture.componentInstance;
+    urlService = TestBed.inject(UrlService);
+    playerPositionService = TestBed.inject(PlayerPositionService);
+    browserCheckerService = TestBed.inject(BrowserCheckerService);
+    explorationPlayerStateService = TestBed.inject(
+      ExplorationPlayerStateService);
+    focusManagerService = TestBed.inject(FocusManagerService);
+    playerTranscriptService = TestBed.inject(PlayerTranscriptService);
+  });
+
+  afterEach(() => {
+    componentInstance.ngOnDestroy();
+  });
+
+  it('should initialize', fakeAsync(() => {
+    let isIframed = true;
+    let mockDisplayedCardIndexChangedEventEmitter = new EventEmitter<void>();
+    let mockOnHelpCardAvailableEventEmitter = (
+      new EventEmitter<HelpCardEventResponse>());
+
+    spyOn(urlService, 'isIframed').and.returnValue(isIframed);
+    spyOn(componentInstance, 'updateDisplayedCardInfo');
+    spyOnProperty(
+      playerPositionService, 'displayedCardIndexChangedEventEmitter')
+      .and.returnValue(mockDisplayedCardIndexChangedEventEmitter);
+    spyOnProperty(playerPositionService, 'onHelpCardAvailable')
+      .and.returnValue(mockOnHelpCardAvailableEventEmitter);
+    componentInstance.ngOnInit();
+    mockDisplayedCardIndexChangedEventEmitter.emit();
+    mockOnHelpCardAvailableEventEmitter.emit({
+      hasContinueButton: true
+    } as HelpCardEventResponse);
+    tick();
+    expect(componentInstance.isIframed).toEqual(isIframed);
+    expect(componentInstance.updateDisplayedCardInfo).toHaveBeenCalled();
+    expect(componentInstance.helpCardHasContinueButton).toBeTrue();
+  }));
+
+  it('should update displayed card info', fakeAsync(() => {
+    let transcriptLength = 10;
+    let displayedCardIndex = 0;
+
+    spyOn(playerTranscriptService, 'getNumCards').and.returnValue(
+      transcriptLength);
+    spyOn(playerPositionService, 'getDisplayedCardIndex').and.returnValue(
+      displayedCardIndex);
+    spyOn(playerTranscriptService, 'isLastCard').and.returnValue(true);
+    spyOn(explorationPlayerStateService, 'isInQuestionMode')
+      .and.returnValue(true);
+    spyOn(focusManagerService, 'setFocusWithoutScroll');
+
+    componentInstance.displayedCard = mockDisplayedCard;
+    componentInstance.updateDisplayedCardInfo();
+    tick(100);
+
+    expect(playerTranscriptService.getNumCards).toHaveBeenCalled();
+    expect(playerPositionService.getDisplayedCardIndex).toHaveBeenCalled();
+    expect(playerTranscriptService.isLastCard).toHaveBeenCalled();
```

Comment:
Is it possible to check any value changes instead of just checking whether the functions are called?

Reply:
done

Reply:
Sorry, If I didn't convey it properly. What I was expecting was to test if any values of a variable like `displayedCardIndex` gets changed after the `updateDisplayedCardInfo` function executes.